pyuvdata/uvdata/uvh5.py

```diff
@@ -447,40 +815,41 @@ def _read_header(
                 "log so that we can fix it."
             )
             assert (
-                np.asarray(header["channel_width"]).size == self.freq_array.size
+                np.asarray(self.channel_width).size == self.freq_array.size
             ), arr_shape_msg
             self._set_future_array_shapes()
 
-        # Pull in the channel_width parameter as either an array or as a single float,
-        # depending on whether or not the data is stored with a flexible spw.
-        if self.flex_spw or np.asarray(header["channel_width"]).ndim == 1:
-            self.channel_width = header["channel_width"][:]
-        else:
-            self.channel_width = float(header["channel_width"][()])
-
-        # get polarization information
-        self.polarization_array = header["polarization_array"][:]
-
-        # get extra_keywords
-        if "extra_keywords" in header:
-            self.extra_keywords = {}
-            for key in header["extra_keywords"].keys():
-                if header["extra_keywords"][key].dtype.type in (np.string_, np.object_):
-                    self.extra_keywords[key] = bytes(
-                        header["extra_keywords"][key][()]
-                    ).decode("utf8")
-                else:
-                    # special handling for empty datasets == python `None` type
-                    if header["extra_keywords"][key].shape is None:
-                        self.extra_keywords[key] = None
-                    else:
-                        self.extra_keywords[key] = header["extra_keywords"][key][()]
+    def _read_header(
+        self,
+        header,
+        filename: str | Path | FastUVH5Meta,
+        run_check_acceptability=True,
+        background_lsts=True,
+    ):
+        """
+        Read header information from a UVH5 file.
+
+        This is an internal function called by the user-space methods.
+        Properties of the UVData object are updated as the file is processed.
 
-        if proc is not None:
-            # if lsts are in the background wait for them to return
-            proc.join()
```

Comment:
I guess the way that you've set up FastUVH5Meta makes it hard to run the lst_array conversion as a process, but it'd be nice not to lose that option when you want to read the whole header. If there's no clean way we can live with this, but running it in a process does help speed things up.

Reply:
Good point. I added it back in.